### PR TITLE
Update video player

### PR DESCRIPTION
**Pull request by benhylau**
- [x] Use IPFS hash as `from` bookmark not timestamp
- [ ] Decide on daily m3u8 naming
- [x] Update [css](https://github.com/tomeshnet/ipfs-live-streaming/tree/master/terraform/shared/video-player/css)
- [x] Fix missing UI elements (picture-in-picture, seekbar) related to above?
- [x] Make share UI pretty
- [x] Hide nav menu underneath for this page? -> 3e77db25eadd73a52313d99e27de65981c09954c

### CODE REVIEW

**Comment by benhylau** [comment]
I only ported video-js.min.css but not common.css is that right?

File: js/video-player-embed.js [hunk 1]
```diff
@@ -1,99 +1,199 @@
...
+
+// Video sharing links
+var rootURL = window.location.href.split('?')[0];
+
+function getShareLink(key) {
+  var bookmark = live.currentTime(); // TODO Change to use ipfs hash
+  return `${rootURL}?m3u8=${m3u8_ipfs}&from=${bookmark}`;
+}
+
+setInterval(function () {
```
Comment (benhylau) [inline js/video-player-embed.js]: @garrying this updates the link visually

File: js/video-player-embed.js [hunk 2]
```diff
@@ -1,99 +1,199 @@
...
+
+if (!stream_urls_http || !Array.isArray(stream_urls_http) || (stream_urls_http.length === 0)) {
+  document.querySelector('.http-stream').setAttribute('disabled', 'disabled');
+}
+
+// Video sharing links
+var rootURL = window.location.href.split('?')[0];
+
+function getShareLink(key) {
+  var bookmark = live.currentTime(); // TODO Change to use ipfs hash
```
Comment (benhylau) [inline js/video-player-embed.js]: @darkdrgn2k this needs to be the ipfs hash

File: js/video-player-embed.js [hunk 3]
```diff
@@ -1,99 +1,199 @@
...
-});
+// Process URL params
+function getURLParam(key) {
+  return new URLSearchParams(window.location.search).get(key);
+}
+
+var ipfs_gw = getURLParam('gw');                          // Set custom IPFS gateway
+if (getURLParam('m3u8'))
+  var m3u8_ipfs = getURLParam('m3u8');                    // Set m3u8 file URL to override IPFS live stream
+var vod_ipfs = getURLParam('vod') || getURLParam('ipfs'); // Set IPFS content hash of mp4 file to play IPFS on-demand video stream ('ipfs' for backward compatability)
```
Comment (benhylau) [inline js/video-player-embed.js]: @darkdrgn2k 2018 uses ipfs=

File: js/video-player-embed.js [hunk 4]
```diff
@@ -1,99 +1,199 @@
...
+  link.value = getShareLink();
+  const tweetURL = link.value;
+  window.open(`https://twitter.com/intent/tweet?url=${encodeURIComponent(tweetURL)}`);
+});
+
+document.querySelector('.share-link').addEventListener('click', function() {
+  var link = document.getElementById('link');
+  link.value = getShareLink();
+  link.select();
+  link.setSelectionRange(0, 99999); // For mobile devices
```
Comment (benhylau) [inline js/video-player-embed.js]: @darkdrgn2k should this be a smaller number?

File: js/video-player-embed.js [hunk 5]
```diff
@@ -1,99 +1,199 @@
...
-
-document.querySelector('.share-link').addEventListener('click', function() {
-  var link = document.getElementById('link');
-  link.value= `${rootURL}?from=${live.currentTime()}&m3u8=live.m3u8`;
-  link.select();
-  link.classList.remove('none')
-  link.setSelectionRange(0, 99999); // For mobile devices
-  document.execCommand('copy');
-  alert('In clipboard');
-});
```
Comment (benhylau) [inline js/video-player-embed.js:99]: @darkdrgn2k I think your files have `^M` that's why all the diff are broken

File: _includes/video-player-embed.html [hunk 6]
```diff
@@ -28,21 +28,19 @@ <h2>Select source</h2>
...
-  <div class="share-container w-100 mt-1">
-    <div class="flex mb-1">
-      <button class="share-link compact">Share link</button>
-      <button class="share-tweet ml-1 compact">Share on Twitter</button>
-    </div>
-    <input type="text" class="input none compact" id="link" />
+  <div class="share-container w-100 mt-1 flex">
+    <button class="share-link mr-1 compact">Share link</button>
+    <button class="share-tweet mr-1 compact">Share on Twitter</button>
+    <input type="text" class="compact" style="flex:1; text-overflow: ellipsis" id="link" />
```
Comment (benhylau) [inline _includes/video-player-embed.html:41]: @garrying sorry I overwrote your changes, but I want the field to show a continually updating hash, can we make it look like this?

File: js/video-player-embed.js (hunk 2 above)
Comment (darkdrgn2k) [reply js/video-player-embed.js]: https://github.com/tomeshnet/ipfs-live-streaming/compare/carriage...ipfsHash?expand=1
`getCurrentHash(live.currentTime())` should do it

File: js/video-player-embed.js (hunk 5 above)
Comment (darkdrgn2k) [reply js/video-player-embed.js:99]: =( thanks for fixing it

**Pull request merged**